ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/EvmToolCommand.java

```diff
@@ -163,23 +186,46 @@ void setBytes(final String optionValue) {
       description = "Number of times to repeat for benchmarking.")
   private final Integer repeat = 0;
 
+  @Option(
+      names = {"-v", "--version"},
+      versionHelp = true,
+      description = "display version info")
+  boolean versionInfoRequested;
+
   static final Joiner STORAGE_JOINER = Joiner.on(",\n");
   private final EvmToolCommandOptionsModule daggerOptions = new EvmToolCommandOptionsModule();
-  private PrintWriter out =
+  PrintWriter out =
       new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out, UTF_8)), true);
+  InputStream in = new ByteArrayInputStream(new byte[0]);
 
-  void parse(final CommandLine.IExecutionStrategy resultHandler, final String[] args) {
+  void execute(final String... args) {
+    execute(System.in, new PrintWriter(System.out, true, UTF_8), args);
+  }
 
-    final CommandLine commandLine = new CommandLine(this);
-    out = commandLine.getOut();
+  void execute(final InputStream input, final PrintWriter output, final String[] args) {
+    final CommandLine commandLine = new CommandLine(this).setOut(output);
+    out = output;
+    in = input;
+
+    // add dagger-injected options
     commandLine.addMixin("Dagger Options", daggerOptions);
 
     // add sub commands here
     commandLine.registerConverter(Address.class, Address::fromHexString);
     commandLine.registerConverter(Bytes.class, Bytes::fromHexString);
     commandLine.registerConverter(Wei.class, arg -> Wei.of(Long.parseUnsignedLong(arg)));
 
-    commandLine.setExecutionStrategy(resultHandler).execute(args);
+    // change negation regexp so --nomemory works
```

Comment:
This might be wrong depending on if we're going for --memory over --nomemory

Reply:
I've added unit tests to verify it works correctly, per the [Beyoncé rule](https://abseil.io/resources/swe-book/html/ch11.html#the_beyonceacutesemicolon_rule)